CLIENTS.md

```diff
@@ -0,0 +1,369 @@
+            // Must match signing algorithm
+            "keyDetails": "PKIX_ECDSA_P256_SHA_256"
+        }
+    }
+}
+
+// Request with an attestation with a self-managed key
+{
+    "dsseRequestV0_0_2": {
+        "envelope": {
+            "payload": "<base64-encoded message>",
+            "payloadType": "<type, e.g. application/vnd.in-toto+json>",
+            "signatures": [
+                {
+                    "sig": "<base64-encoded signature>",
+                    "keyid": ""
+                }
+            ]
+        },
+        "verifier": {
+            "publicKey": {
+                "rawBytes": "<base64 DER-encoded public key>"
+            },
+            // Must match signing algorithm
+            "keyDetails": "PKIX_ECDSA_P256_SHA_256"
+        }
+    }
+}
+```
+
+The response will be a
+[`TransparencyLogEntry` message](https://github.com/sigstore/protobuf-specs/blob/5296f13d62e7fad428581d969f664c30cc52f549/protos/sigstore_rekor.proto#L94),
+which should be persisted in a bundle. Clients no longer need to transform the
+Rekor response into a `TLE` message to store in the bundle.
+
+### Two Entry Types
+
+Rekor v2 only supports `hashedrekord` (`HashedRekordRequestV0_0_2`) and
+`dsse` (`DSSERequestV0_0_2`) entry types, dropping a number of unused types
+such as `jar`, `alpine`, `rpm`, and the older types `rekord` and `intoto`.
+Additional types may be added in the future if there is demand, but this
+will require updating the client specification so that all clients implement
+support for these types.
+
+### Certificate and Public Key Verifiers
+
+Rekor v2 only supports signature verification using a certificate or a
+public key, dropping support for PGP, minisign, pkcs7, SSH and TUF.
+Additional verifiers may be added in the future, but this will also require
+updating the client specification.
+
+### Handling Longer Requests
+
+Clients need to increase request timeouts to at least 10 seconds.
+
+Rekor now batches uploads so that checkpoints are published less frequently.
+Additionally, we plan to support synchronous witnessing, where third-party
+witnesses independently verify the consistency of the log and Rekor provides
+co-signed checkpoints with each upload response.
+
+If a client needs to create multiple entries, it is recommended to upload those
+entries in parallel.
+
+## TrustedRoot lookup by checkpoint key ID rather than log ID
+
+Log ID, the SHA-256 digest of the log's public key, is used as a "unique"
+identifier for a log. The TransparencyLogEntry message includes it in a
+bundle, and clients use that log ID to lookup the correct log public key
+in the TrustedRoot to verify the bundle.
+
+Rekor will no longer include log IDs in the response, and instead clients
+should use the checkpoint key ID as specified in the
+[C2SP spec](https://github.com/C2SP/C2SP/blob/main/signed-note.md#signatures)
+to lookup the correct log public key to verify a bundle. Each transparency
```

Comment:
I think this might not be clear enough like this. How exactly do clients calculate the checkpoint key id for an entry -- is entrys checkpoint origin the "key name"?
 
 is there an example implementation of this?

Reply:
Added selection algorithms for this, thanks. The `TransparencyLogInstance` proto has been updated as well to document this.

tl;dr either 1) compare the bundle's `log_id` to the trust root log instance's `checkpoint_key_id` or `log_id` (what most clients are doing already), 2) enumerate all log instances and compare the checkpoint's key ID from the signature line to the first 4 bytes of the trust root log instance's `checkpoint_key_id`/`log_id` (which is what sigstore-python is doing), or 3) compute the log instance's checkpoint key ID using the instance's public key and origin and compare (this is Hard Mode, and I doubt any client will)